src/main/java/org/opensearch/neuralsearch/highlight/SemanticHighlighter.java

```
@@ -60,6 +60,12 @@ public HighlightField highlight(FieldHighlightContext fieldContext) {
             return null;
         }
 
+        String[] preTags = fieldContext.field.fieldOptions().preTags();
+        String[] postTags = fieldContext.field.fieldOptions().postTags();
+        if (preTags != null && preTags.length > 0 && postTags != null && postTags.length > 0) {
+            semanticHighlighterEngine = semanticHighlighterEngine.withTags(preTags[0], postTags[0]);
```

Shouldn't we left a log for error case where only of of tags are provided?

---

I think we should throw IllegalArgumentException if invalid custom tags are provided rather than silently ignore it.

Other than validate both pre and post tags should be provided and they should not be empty string is there a way to also validate the tag should be a valid tag that we can process? And also validate the length of the tags to ensure user cannot provide an extremely long string which may break our system?

---

Updated to not have validation here, since it always by default will pass in <em> and </em> from core highlighter.

Since it's customized tags controlled by user, we can't know what is invalid or not.